lib/sign-in-component.js

```diff
@@ -70,10 +78,31 @@ class SignInComponent {
       : null
   }
 
+  attachChangeEvent () {
+    const previousTokenEditor = this.editor
+    this.editor = this.refs.editor
+
+    if (!previousTokenEditor && this.editor) {
+      this.editor.onDidChange(() => {
+        const token = this.refs.editor.getText().trim()
+        this.refs.loginButton.disabled = !token
+      })
+    }
+  }
+
   async signIn () {
+    const token = this.refs.editor.getText().trim()
+    
+    if (!token) {
+      this.props.notificationManager.addError('Invalid login token', {
```

Comment:
Can we use `renderErrorMessage` here instead of a notification?

If we use a notification, that notification will appear all the way in the upper-right corner of the app. If we use `renderErrorMessage`, we can show the error message right next to the token field (i.e., right next to the error 🌟 ).